src/api/source/geojson.js

```diff
@@ -0,0 +1,184 @@
+import Base from './base';
+import { Dataframe } from '../../core/renderer';
+import * as rsys from '../../client/rsys';
+import * as util from '../util';
+
+
+// TODO: add docs / API
+
+export default class GeoJSON extends Base {
+
+    constructor(data) {
+        super();
+        data = data || {};
+        if (data.type === 'FeatureCollection') {
+            this._features = data.features || [];
+        } else if (data.type === 'Feature') {
+            this._features = [data];
+        }
+        else {
+            throw Error('No valid GeoJSON data');
+        }
+        this._status = 'init'; // init -> metadata -> data
+        this._type = ''; // Point, LineString, MultiLineString, Polygon, MultiPolygon
+    }
+
+    bindLayer(addDataframe, removeDataframe) {
+        this._addDataframe = addDataframe;
+        this._removeDataframe = removeDataframe;
+    }
+
+    requestData() {
```

Comment:
this method is very confusing to me, to many responsabilities, sometimes returning a promise, global state...

Reply:
Totally agree. That's why I added the comment to refactor it (for a future PR)